app/controllers/todos_controller.rb

```diff
@@ -1 +1,81 @@
-# This controller is for all the CRUD operations related to a Todo.
\ No newline at end of file
+MyApp.get "/todos/new" do
+  @current_user = User.find_by_id(session["user_id"]) 
+    if @current_user != nil
+      erb :"todos/new"
+    else
+      erb :"users/log_in_first"
+    end
+end
+
+MyApp.post "/todos/create_item" do
+  @current_user = User.find_by_id(session["user_id"]) 
+    if @current_user != nil
+      @todos = Todo.new
+      @todos.title = params["title"]
+      @todos.description = params["description"]
+      @todos.completed = params["completed"]
+      @todos.user_id = params["user_id"]
+      @todos.save
+      erb :"todos/list_item_created"
+    else
+      erb :"users/log_in_first"
+    end
+end
+
+MyApp.get "/todos/list" do
+  @current_user = User.find_by_id(session["user_id"]) 
+    if @current_user != nil
+      @todos = Todo.all
+      erb :"todos/todo_list"
+    else
+      erb :"users/log_in_first"
+    end
+end
+
+MyApp.get "/todos/edit_todo" do
+  @current_user = User.find_by_id(session["user_id"]) 
+    if @current_user != nil
+      @todo = Todo.find_by_id(session["user_id"])
```

Comment:
`Todo.find_by_id(session["user_id"])` is fetching a row from the 'todos' table that has an ID equal to the **current user's** ID. You should instead be fetching the todo row based on some ID from `params`.